Sources/FluidAudio/Diarizer/README.md

```diff
@@ -0,0 +1,310 @@
+# Speaker Diarization
+
+Real-time speaker diarization for iOS and macOS, answering "who spoke when" in audio streams.
+
+## Overview
+
+The FluidAudio Diarizer provides production-ready speaker diarization with:
+- **17.7% DER** on AMI corpus (competitive with state-of-the-art)
+- **150+ RTFx** real-time performance on Apple Silicon
+- **On-device processing** using Core ML
+- **Streaming support** for real-time applications
+
+## Quick Start
+
+```swift
+import FluidAudio
+
+// 1. Download models (one-time setup)
+let models = try await DiarizerModels.downloadIfNeeded()
+
+// 2. Configure and initialize
+let config = DiarizerConfig(
+    clusteringThreshold: 0.7,  // Optimal for best accuracy
+    minSpeechDuration: 1.0,     // Minimum speech segment
+    minSilenceGap: 0.5          // Minimum gap between speakers
+)
+
+let diarizer = DiarizerManager(config: config)
+diarizer.initialize(models: models)
+
+// 3. Process audio
+let audioSamples: [Float] = loadAudioFile() // 16kHz mono
+let result = try diarizer.performCompleteDiarization(audioSamples)
+
+// 4. Get results
+for segment in result.segments {
+    print("Speaker \(segment.speakerId): \(segment.startTimeSeconds)s - \(segment.endTimeSeconds)s")
+}
+```
+
+## Core Components
+
+### DiarizerManager
+Main entry point for diarization pipeline:
+```swift
+let diarizer = DiarizerManager(config: config)
+diarizer.initialize(models: models)
+let result = try diarizer.performCompleteDiarization(audio)
+```
+
+### SpeakerManager
+Tracks speaker identities across audio chunks:
+```swift
+let speakerManager = diarizer.speakerManager
+
+// Get speaker information
+print("Active speakers: \(speakerManager.speakerCount)")
+for speakerId in speakerManager.speakerIds {
+    if let info = speakerManager.getSpeakerInfo(for: speakerId) {
+        print("\(speakerId): \(info.totalDuration)s total")
+    }
+}
+```
+
+### DiarizerConfig
+Configuration parameters:
+```swift
+let config = DiarizerConfig(
+    clusteringThreshold: 0.7,      // Speaker separation threshold (0.0-1.0)
+    minSpeechDuration: 1.0,         // Minimum speech duration in seconds
+    minSilenceGap: 0.5,             // Minimum silence between speakers
+    minActiveFramesCount: 10.0,     // Minimum active frames for valid segment
+    debugMode: false                // Enable debug logging
+)
+```
+
+## Streaming/Real-time Processing
+
+Process audio in chunks for real-time applications:
+
+```swift
+// Configure for streaming
+let diarizer = DiarizerManager(
+    config: config,
+    chunkDuration: 10.0,    // Process in 10-second chunks
+    chunkOverlap: 0.0        // No overlap for lowest latency
+)
+diarizer.initialize(models: models)
+
+// Process stream
+for audioChunk in audioStream {
+    let result = try diarizer.performCompleteDiarization(audioChunk)
+    
+    // Real-time results
+    for segment in result.segments {
+        handleSpeakerSegment(segment)
+    }
```

Comment:
How does this work if its 10 second chunks, will audIOChunk need to be 10s long?